### DISCUSSION TITLE
Fetching messages from remote source after build

### DISCUSSION THREAD

**Discussion by cnysten-tieto** [amannn/next-intl#585] [Q&A]
Hello!

First of all I want to say thanks to all maintainers and the whole next-intl community for a great piece of software, I really appreciate the work you are all putting in to it. 🙏🏻

My question is about fetching message json files from a remote source (S3) for use in Client Components. Please bear with me if the answer to my question is obvious, I'm new to more or less everything in the project, so it might be that I'm struggling with something fairly straight forward. I tried to look for similar discussions in issues and here in discussions, but wasn't able to find anything that would answer this question.

Initially we had the json files locally in the repo but then switched to fetching them from S3. This lead to the build being dependent on access to S3, which caused a few issues. So my question is there a way to fetch the messages only after the build? And how would one go about it?

Current setup is essentially the same as in this example from the next-intl docs. The only difference really is that the file is fetched and not imported.
```
import {NextIntlClientProvider} from 'next-intl';
import {notFound} from 'next/navigation';
 
export function generateStaticParams() {
  return [{locale: 'en'}, {locale: 'de'}];
}
 
export default async function LocaleLayout({children, params: {locale}}) {
  let messages;
  try {
    messages = (await import(`../../messages/${locale}.json`)).default;
  } catch (error) {
    notFound();
  }
 
  return (
    <html lang={locale}>
      <body>
        <NextIntlClientProvider locale={locale} messages={messages}>
          {children}
        </NextIntlClientProvider>
      </body>
    </html>
  );
}
```

Any help or pointers are greatly appreciated!

**Comment by bryanjtc** [comment]
> Hello!
> 
> First of all I want to say thanks to all maintainers and the whole next-intl community for a great piece of software, I really appreciate the work you are all putting in to it. 🙏🏻
> 
> My question is about fetching message json files from a remote source (S3) for use in Client Components. Please bear with me if the answer to my question is obvious, I'm new to more or less everything in the project, so it might be that I'm struggling with something fairly straight forward. I tried to look for similar discussions in issues and here in discussions, but wasn't able to find anything that would answer this question.
> 
> Initially we had the json files locally in the repo but then switched to fetching them from S3. This lead to the build being dependent on access to S3, which caused a few issues. So my question is there a way to fetch the messages only after the build? And how would one go about it?
> 
> Current setup is essentially the same as in this example from the next-intl docs. The only difference really is that the file is fetched and not imported.
> ```
> import {NextIntlClientProvider} from 'next-intl';
> import {notFound} from 'next/navigation';
>  
> export function generateStaticParams() {
>   return [{locale: 'en'}, {locale: 'de'}];
> }
>  
> export default async function LocaleLayout({children, params: {locale}}) {
>   let messages;
>   try {
>     messages = (await import(`../../messages/${locale}.json`)).default;
>   } catch (error) {
>     notFound();
>   }
>  
>   return (
>     <html lang={locale}>
>       <body>
>         <NextIntlClientProvider locale={locale} messages={messages}>
>           {children}
>         </NextIntlClientProvider>
>       </body>
>     </html>
>   );
> }
> ```
> 
> Any help or pointers are greatly appreciated!

Did you found a solution?

**Reply by cnysten-tieto** [reply]
Not really in the sense I was thinking when writing the original question, but we did find a solution that solved our problems.

We ended up adding a script that replaces the different json files during the build process. This happens before `next build` and only in our cloud builds. This way when building locally there is no dependency on the remote files, but in our production and dev environments running in the cloud we still are able to use the correct files for each environment.

Hope that helps!
